js_modules/dagit/packages/core/src/metadata/MetadataEntry.tsx

```diff
@@ -171,6 +182,51 @@ export const MetadataEntry: React.FC<{
       return null;
     case 'TableSchemaMetadataEntry':
       return <TableSchema schema={entry.schema} />;
+    case 'NotebookMetadataEntry':
+      if (repoLocation) {
```

Comment:
One thing i'd like to do is change this so that it supports URLs like in the ipynb plugin ([here](https://github.com/dagster-io/dagster/blob/master/js_modules/dagit/packages/core/src/plugins/ipynb.tsx#L22)) but i'm not sure how to do the conditional check (with `React.useMemo`). if i put it in this condition i get a linting error that `useMemo` can't be called conditionally. But putting it at the top of the function also seems wrong since we don't know what attribute of `entry` to access or what type it will be

Reply:
Hey @jamiedemaria! Ahh ok - what we should probably do here is take the content of `SidebarComponent.tsx` that handles the URL + rendering of the notebook and move it to a separate component. It looks like that SIdebarComponent does a /bit/ of unwrapping, etc. in the first few lines, but then it's mostly the same as what we want here?

You can only use React hooks like useEffect and useMemo before the first if/return statement of your component, so it'll complain about one being inside a switch here. Once we have that component that is shared between SidebarComponent and this MetadataEntry, we can say `return <NotebookButton repoLocation={repoLocation} path={entry.path} />` in both places and we should be set!

Reply:
cool! that sounds good